Sources/SwiftKafka/KafkaClient.swift

```diff
@@ -83,18 +86,64 @@ final class KafkaClient: Sendable {
         }
     }
 
-    /// Polls the Kafka client for events.
-    ///
-    /// Events will cause application-provided callbacks to be called.
+    /// Swift wrapper for events from `librdkafka`'s event queue.
+    enum RDKafkaEvent {
+        case deliveryReport(results: [Result<KafkaAcknowledgedMessage, KafkaAcknowledgedMessageError>])
+    }
+
+    /// Poll the event `rd_kafka_queue_t` for new events.
     ///
-    /// - Parameter timeout: Specifies the maximum amount of time
-    /// (in milliseconds) that the call will block waiting for events.
-    /// For non-blocking calls, provide 0 as `timeout`.
-    /// To wait indefinitely for an event, provide -1.
-    /// - Returns: The number of events served.
-    @discardableResult
-    func poll(timeout: Int32) -> Int32 {
-        return rd_kafka_poll(self.kafkaHandle, timeout)
+    /// - Parameter maxEvents:Maximum number of events to serve in one invocation.
+    func eventPoll(maxEvents: Int = 1000) -> [RDKafkaEvent] {
+        var events = [RDKafkaEvent]()
+
+        getEventsLoop: for _ in 0..<maxEvents {
```

Comment:
I am not a big fan of named loops. I don't think we need this here since we can just return in the case where we call `break getEventsLoop` right now.